### DISCUSSION TITLE
There are a lot of warnings in version 4.9(Angular13)

### DISCUSSION THREAD

**Discussion by AllForNothing** [swagger-api/swagger-ui#7961] [Q&A] [answered]
I'm using Swagger-UI in an Angular13 project, after upgrading Swagger-UI to 4.9, I get a lot of warnings in my console when running npm start.   To resolve these warnings, I have to add the below content 
`"core-js-pure",
"lodash",
"prop-types",
"randexp",
"react-copy-to-clipboard",
"react-debounce-input",
"react-immutable-proptypes",
"redux-immutable",
"url-parse",
"xml",
"xml-but-prettier",
"sha.js"` 
to the anglar.json file's `allowedCommonJsDependencies` array.  
This does solve the problem, but it's a little inconvenient.
Does anyone know the reason for these warnings please? Or is there a better way to get rid of the warnings?

Thanks!!!

 

`Warning: node_modules/@babel/runtime-corejs3/core-js/array/from.js depends on 'core-js-pure/features/array/from'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/array/is-array.js depends on 'core-js-pure/features/array/is-array'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/get-iterator-method.js depends on 'core-js-pure/features/get-iterator-method'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/instance/bind.js depends on 'core-js-pure/features/instance/bind'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/instance/filter.js depends on 'core-js-pure/features/instance/filter'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/instance/for-each.js depends on 'core-js-pure/features/instance/for-each'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/instance/index-of.js depends on 'core-js-pure/features/instance/index-of'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/instance/slice.js depends on 'core-js-pure/features/instance/slice'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/map.js depends on 'core-js-pure/features/map'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/assign.js depends on 'core-js-pure/features/object/assign'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/create.js depends on 'core-js-pure/features/object/create'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/define-properties.js depends on 'core-js-pure/features/object/define-properties'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/define-property.js depends on 'core-js-pure/features/object/define-property'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/get-own-property-descriptor.js depends on 'core-js-pure/features/object/get-own-property-descriptor'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/get-own-property-descriptors.js depends on 'core-js-pure/features/object/get-own-property-descriptors'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/get-own-property-symbols.js depends on 'core-js-pure/features/object/get-own-property-symbols'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/get-prototype-of.js depends on 'core-js-pure/features/object/get-prototype-of'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/keys.js depends on 'core-js-pure/features/object/keys'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/object/set-prototype-of.js depends on 'core-js-pure/features/object/set-prototype-of'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/promise.js depends on 'core-js-pure/features/promise'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/reflect/construct.js depends on 'core-js-pure/features/reflect/construct'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/reflect/get.js depends on 'core-js-pure/features/reflect/get'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/symbol.js depends on 'core-js-pure/features/symbol'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/@babel/runtime-corejs3/core-js/symbol/iterator.js depends on 'core-js-pure/features/symbol/iterator'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/camelCase'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/debounce'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/find'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/get'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning:node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/isEmpty'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/isFunction'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/isString'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning:node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/memoize'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning:node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/merge'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/omit'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/reduce'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/set'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/some'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/toLower'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/toString'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/upperFirst'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'lodash/zipObject'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'prop-types'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'randexp'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'react-copy-to-clipboard'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'react-debounce-input'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'react-immutable-proptypes'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'redux-immutable'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'sha.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'url-parse'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'xml'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: node_modules/swagger-ui/dist/swagger-ui-es-bundle-core.js depends on 'xml-but-prettier'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies`

**Comment by barrywilks7** [comment, marked as answer]
see https://github.com/swagger-api/swagger-ui/issues/6678
and https://github.com/TimUnderhay/swagger-ui-es

**Reply by AllForNothing** [reply]
Thanks, it solved my confusion.
